diff --git a/modules/upnp/SCsub b/modules/upnp/SCsub
index 6657d75cae..a225311475 100644
--- a/modules/upnp/SCsub
+++ b/modules/upnp/SCsub
@@ -35,6 +35,7 @@ if env["builtin_miniupnpc"]:
         env_upnp.Append(CPPDEFINES=["MINIUPNPC_SET_SOCKET_TIMEOUT"])
 
     env_thirdparty = env_upnp.Clone()
+    env_thirdparty.Prepend(CPPPATH=[thirdparty_dir + "include/miniupnpc"])
     env_thirdparty.disable_warnings()
     env_thirdparty.add_source_files(thirdparty_obj, thirdparty_sources)
     env.modules_sources += thirdparty_obj
diff --git a/modules/upnp/upnp.cpp b/modules/upnp/upnp.cpp
index 4305bf842a..de8cdf5ec4 100644
--- a/modules/upnp/upnp.cpp
+++ b/modules/upnp/upnp.cpp
@@ -30,8 +30,8 @@
 
 #include "upnp.h"
 
-#include <miniwget.h>
-#include <upnpcommands.h>
+#include <miniupnpc/miniwget.h>
+#include <miniupnpc/upnpcommands.h>
 
 #include <stdlib.h>
 
diff --git a/modules/upnp/upnp.h b/modules/upnp/upnp.h
index dc9bbdbc22..7753b0d30a 100644
--- a/modules/upnp/upnp.h
+++ b/modules/upnp/upnp.h
@@ -35,7 +35,7 @@
 
 #include "core/object/ref_counted.h"
 
-#include <miniupnpc.h>
+#include <miniupnpc/miniupnpc.h>
 
 class UPNP : public RefCounted {
 	GDCLASS(UPNP, RefCounted);
diff --git a/modules/upnp/upnp_device.cpp b/modules/upnp/upnp_device.cpp
index 11ee3681af..149900d590 100644
--- a/modules/upnp/upnp_device.cpp
+++ b/modules/upnp/upnp_device.cpp
@@ -32,7 +32,7 @@
 
 #include "upnp.h"
 
-#include <upnpcommands.h>
+#include <miniupnpc/upnpcommands.h>
 
 String UPNPDevice::query_external_address() const {
 	ERR_FAIL_COND_V_MSG(!is_valid_gateway(), "", "The Internet Gateway Device must be valid.");
diff --git a/thirdparty/miniupnpc/include/igd_desc_parse.h b/thirdparty/miniupnpc/include/miniupnpc/igd_desc_parse.h
similarity index 100%
rename from thirdparty/miniupnpc/include/igd_desc_parse.h
rename to thirdparty/miniupnpc/include/miniupnpc/igd_desc_parse.h
diff --git a/thirdparty/miniupnpc/include/miniupnpc.h b/thirdparty/miniupnpc/include/miniupnpc/miniupnpc.h
similarity index 100%
rename from thirdparty/miniupnpc/include/miniupnpc.h
rename to thirdparty/miniupnpc/include/miniupnpc/miniupnpc.h
diff --git a/thirdparty/miniupnpc/include/miniupnpc_declspec.h b/thirdparty/miniupnpc/include/miniupnpc/miniupnpc_declspec.h
similarity index 100%
rename from thirdparty/miniupnpc/include/miniupnpc_declspec.h
rename to thirdparty/miniupnpc/include/miniupnpc/miniupnpc_declspec.h
diff --git a/thirdparty/miniupnpc/include/miniupnpctypes.h b/thirdparty/miniupnpc/include/miniupnpc/miniupnpctypes.h
similarity index 100%
rename from thirdparty/miniupnpc/include/miniupnpctypes.h
rename to thirdparty/miniupnpc/include/miniupnpc/miniupnpctypes.h
diff --git a/thirdparty/miniupnpc/include/miniwget.h b/thirdparty/miniupnpc/include/miniupnpc/miniwget.h
similarity index 100%
rename from thirdparty/miniupnpc/include/miniwget.h
rename to thirdparty/miniupnpc/include/miniupnpc/miniwget.h
diff --git a/thirdparty/miniupnpc/include/portlistingparse.h b/thirdparty/miniupnpc/include/miniupnpc/portlistingparse.h
similarity index 100%
rename from thirdparty/miniupnpc/include/portlistingparse.h
rename to thirdparty/miniupnpc/include/miniupnpc/portlistingparse.h
diff --git a/thirdparty/miniupnpc/include/upnpcommands.h b/thirdparty/miniupnpc/include/miniupnpc/upnpcommands.h
similarity index 100%
rename from thirdparty/miniupnpc/include/upnpcommands.h
rename to thirdparty/miniupnpc/include/miniupnpc/upnpcommands.h
diff --git a/thirdparty/miniupnpc/include/upnpdev.h b/thirdparty/miniupnpc/include/miniupnpc/upnpdev.h
similarity index 100%
rename from thirdparty/miniupnpc/include/upnpdev.h
rename to thirdparty/miniupnpc/include/miniupnpc/upnpdev.h
diff --git a/thirdparty/miniupnpc/include/upnpreplyparse.h b/thirdparty/miniupnpc/include/miniupnpc/upnpreplyparse.h
similarity index 100%
rename from thirdparty/miniupnpc/include/upnpreplyparse.h
rename to thirdparty/miniupnpc/include/miniupnpc/upnpreplyparse.h
